catalog/core/catalog-core-localstorageprovider/src/main/java/org/codice/ddf/catalog/content/impl/FileSystemStorageProvider.java

```diff
@@ -564,47 +581,52 @@ private ContentItem generateContentFile(
 
     Path contentItemPath =
         Paths.get(contentDirectory.toAbsolutePath().toString(), item.getFilename());
-    ByteSource byteSource;
-
-    long copy;
+    byte[] plainBytes;
```

Comment:
❗️ This might be too large to hold in memory for one file or multiple files concurrently.  See previous uses of `ByteSource` in this class.